ykman/piv.py

```diff
@@ -585,6 +588,17 @@ def set_mgm_key(self, new_key, touch=False, store_on_device=False):
         if store_on_device and not new_key:
             new_key = generate_random_management_key()
 
+        if store_on_device or (not store_on_device and self.has_stored_key):
+            # Ensure we have access to protected data before overwriting key
+            try:
+                self._init_pivman_protected()
+            except Exception as e:
+                logger.debug('Failed to initialize protected pivman data',
+                             exc_info=e)
+
+                if store_on_device:
+                    raise e
```

Comment:
Any reason for `raise e` and not just an empty `raise` here ? I think we usually just use `rasie` so you might want to change to that for consistency.

Reply:
No reason other than me not being used to bare `raise` statements being a thing. :)